mbed_greentea/mbed_greentea_cli.py

```diff
@@ -833,7 +833,9 @@ def main_cli(opts, args, gt_instance_uuid=None):
         # Reports (to file)
         if opts.report_junit_file_name:
             gt_logger.gt_log("exporting to JUnit file '%s'..."% gt_logger.gt_bright(opts.report_junit_file_name))
-            junit_report = exporter_testcase_junit(test_report, test_suite_properties = get_test_suite_properties())
+            # This test specification will be used by JUnit exporter to populate TestSuite.properties (useful meta-data for Viewer)
+            junit_test_spec = test_spec if opts.test_spec else None
+            junit_report = exporter_testcase_junit(test_report, test_suite_properties = get_test_suite_properties(test_spec=junit_test_spec))
```

Comment:
Minor gripe: can you flatten this line out a bit? It's kinda hard to read.
```
junit_report = exporter_testcase_junit(test_report, test_suite_properties = get_test_suite_properties(test_spec=junit_test_spec))
```

Maybe something like this?
```
test_suite_properties = get_test_suite_properties(test_spec=junit_test_spec)
junit_report = exporter_testcase_junit(test_report, test_suite_properties = test_suite_properties)
```